libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -104,7 +104,10 @@ uint16_t AP_Arming::compass_magfield_expected() const
 
 bool AP_Arming::is_armed()
 {
-    return require == NONE || armed;
+    if ( require == NONE && !armed ) {
+        arm(NONE);
```

Comment:
Sorry, I don't think this is the correct way to do things.

There's no way that calling `is_armed()` should arm the aircraft; it should really be a `const` method, but various caching and filters mean it can't be IIRC.

Reply:
I got your point here, do you have suggestion where to it do it?
I did this as arming_require=0 means I really want is_armed to be always true and it was already implemented in this function.